webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.js

```diff
@@ -12,9 +12,18 @@ class BreadcrumbBar extends React.Component {
       loadSwitcherResourcesByResource,
       currentPage,
       resourceUrl,
+      resourceSwitcherItems,
     } = this.props;
-
-    if (!currentPage || resourceUrl !== resource.resourceUrl) {
+    const isCorrectList =
+      resourceSwitcherItems.length > 0
+        ? resourceSwitcherItems[0].url ===
+          resource.switcherItemUrl.replace(':id', resourceSwitcherItems[0].id)
```

Comment:
this is pretty  inefficient, how about move this logic to a selector with memorization (`reselect`)?
we have `reselect` already, you can read about it [here](https://github.com/reduxjs/reselect)

Reply:
Because the resource is coming from the server it doesn't work well in a selector.